### PR TITLE
Add generation of common protos pkgs

### LINKED ISSUE #9: Generate packages containing the messages of well-known googleapis protos
Packages need to be generated for each language supported by api-packager that contain the code generated from the well-known googleapis protos.

Rationale:
- [grpc-api-packager][] is used to create gRPC service packages from protos in the [googleapis][] project
- Some of the gRPC services depend on well-known photos
  - some well-known protos, e.g [empty.proto][] are defined in the [protobuf][] package and have copies in [googleapis][]
  - others, e.g. [operations.proto][] are exclusive to [googleapis][]
- It's possible to update [grpc-api-packager][] to include generated code for any well-known proto it depends on and include it in the generated package
- This leads to potential conflicts if two different generated libraries that depend on the same well-known proto are imported.
- This is not an issue in all generated libraries
  - e.g, its not an issue for the nodeJS generated libraries because of how the client constructor is generated dynamically using ProtoBuf.js
- It is an issue for most other packaged languages (ruby, python, php, c#, objective-c, go)
- In all these languages, the proposal is to
  - generate a package that contains generated messages for all the well-known protos in googleapis
  - make this a dependency of all packages whose services depend on well-known protos

Summary:
- these packages should also be generated by [grpc-api-packager][], probably with a different set of templates in the various language to reflect differences between these packages and actual service packages. 
- a candidate name for the package is `grpc-googleapis-messages`
- the package's version would be linked to the version of a release branch of the googleapis repo

[googleapis]: https://github.com/google/googleapis
[protobuf]: https://github.com/google/protobuf
[operations.proto]: https://github.com/google/googleapis/blob/master/google/longrunning/operations.proto
[empty.proto]: https://github.com/google/googleapis/blob/master/google/protobuf/empty.proto
[grpc-api-packager]: https://github.com/grpc/grpc-tools/blob/master/bin/grpc-api-packager

### PR DESCRIPTION
**Pull request by tbetbetbe**
This is based on the changes #17 and should only be committed after that PR is accepted.
However the changes here can be reviewed at the same time.

Fixes #9

### CODE REVIEW

File: bin/grpc-api-packager [hunk 1]
```diff
@@ -69,29 +82,33 @@ var parseArgs = function parseArgs() {
...
+      nargs: '*',
+      dest: 'languages'
     }
   );
   cli.addArgument(
     [ '-r', '--services_root' ],
     {
       help: 'The directory containing the rpc service proto definitions.'
-          + '\nShould not be specified with -r/--protobuf, which clobbers this.'
+          + '\nShould not be specified with -r/--protobuf, which clobbers this.',
```
Comment (saicheems) [inline bin/grpc-api-packager]: Nit: some help strings end with a period, some don't.

File: bin/grpc-api-packager [hunk 2]
```diff
@@ -51,17 +52,29 @@ var parseArgs = function parseArgs() {
...
+        ' outside of an apis directory, but within services_root or repository',
+      dest: 'includePath'
     }
   );
   cli.addArgument(
-    [ '-o', '--out_path' ],
+    [ '-o', '--out_dir' ],
     {
-      help: 'The directory in which the generated packages will be saved.'
+      help: 'The directory in which the generated packages will be saved.',
```
Comment (saicheems) [inline bin/grpc-api-packager]: Nit: period.

File: bin/grpc-api-packager [hunk 3]
```diff
@@ -51,17 +52,29 @@ var parseArgs = function parseArgs() {
     version: version
   });
   cli.addArgument(
+    [ '-x', '--build_common_protos' ],
+    {
+      defaultValue: false,
+      action: 'storeTrue',
+      help: 'When set indicates that package for the common api'
+            + ' protos should be built.\n',
```
Comment (saicheems) [inline bin/grpc-api-packager]: Same here

File: lib/api_repo.js [hunk 4]
```diff
@@ -119,11 +148,12 @@ ApiRepo.prototype.buildPackages =
...
       if (packager[l]) {
         var buildAPackage = function buildAPackage(generated, next) {
-          var opts = {
+          var opts = _.merge({
             'top': path.join(that.outDir, l),
             'packageInfo': config.defaults,
             'generated': generated
-          };
+          }, templateDirs[l]);
+          console.log('what are opts %j', opts);
```
Comment (saicheems) [inline lib/api_repo.js]: Leftover log?

File: lib/config.js [hunk 5]
```diff
@@ -4,10 +4,21 @@ var fs = require('fs');
...
+var defaultDepPath = configPath('dependencies.yml');
+var defaultApiPath = configPath('api_defaults.yml');
+var defaultCommonProtosPath = configPath('common_protos.yml');
 
 exports.defaults = {
-  api: yaml.safeLoad(fs.readFileSync(defaultApiPath, 'utf8')),
-  dependencies:  yaml.safeLoad(fs.readFileSync(defaultDepPath, 'utf8'))
-}
+  api: loadYamlSync(defaultApiPath),
+  dependencies:  loadYamlSync(defaultDepPath)
```
Comment (saicheems) [inline lib/config.js]: Nit: 1 space after :

File: templates/commonpb/ruby/PUBLISHING.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+= PUBLISHING
```
Comment (saicheems) [inline templates/commonpb/ruby/PUBLISHING.md]: Nit: Github markdown uses # for headings

**Comment by saicheems** [comment]
Style nits mostly, but the code itself lgtm on inspection.

**Comment by tbetbetbe** [comment]
Can repeat any comments that affect changes in #17 on that PR please ?

File: bin/grpc-api-packager (hunk 1 above)
Comment (tbetbetbe) [inline bin/grpc-api-packager]: Fixed

File: bin/grpc-api-packager (hunk 2 above)
Comment (tbetbetbe) [inline bin/grpc-api-packager]: Fixed

File: templates/commonpb/ruby/PUBLISHING.md (hunk 6 above)
Comment (tbetbetbe) [inline templates/commonpb/ruby/PUBLISHING.md]: Fixed

File: lib/config.js (hunk 5 above)
Comment (tbetbetbe) [inline lib/config.js]: Fixed

File: bin/grpc-api-packager (hunk 3 above)
Comment (tbetbetbe) [inline bin/grpc-api-packager]: Fixed

**Pull request merged**